bindings/py/cpp_src/bindings/engine/py_Engine.cpp

```diff
@@ -241,56 +242,77 @@ namespace htm_ext
+                py::buffer_info info = b.request();  /* Request a buffer descriptor from Python */
+                if (info.ndim != 1)
+                    throw std::runtime_error("Expected a one dimensional array!");
+                size_t size = static_cast<size_t>(info.shape[0]);
+                NTA_BasicType type;
+                if      (((info.format == "i") || (info.format == "l") ) && info.itemsize == 4) type = NTA_BasicType_Int32;
+                else if (((info.format == "I") || (info.format == "L") ) && info.itemsize == 4) type = NTA_BasicType_UInt32;
+                else if ((info.format == "l") || (info.format == "q") ) type = NTA_BasicType_Int64;
+                else if ((info.format == "L") || (info.format == "Q") ) type = NTA_BasicType_UInt64;
+                else if (info.format == "f") type = NTA_BasicType_Real32;
+                else if (info.format == "d") type = NTA_BasicType_Real64;
+                else if (info.format == py::format_descriptor<bool>::format()) type = NTA_BasicType_Bool;
+                else if (info.format == py::format_descriptor<Byte>::format()) type = NTA_BasicType_Byte;
+                else NTA_THROW << "setInputArray(): Unexpected data type in the array!  info.format=" << info.format;
+                // for info.format codes, see https://docs.python.org/3.7/library/array.html
+                Array s(type, info.ptr, size);
+                std::cout << "src: " << s << std::endl;
+
+                r.setInputData(name, s);
+            });
+
+
+        py_Region.def("getParameterInt32", &Region::getParameterInt32)
+            .def("getParameterUInt32", &Region::getParameterUInt32)
+            .def("getParameterInt64",  &Region::getParameterInt64)
+            .def("getParameterUInt64", &Region::getParameterUInt64)
+            .def("getParameterReal32", &Region::getParameterReal32)
+            .def("getParameterReal64", &Region::getParameterReal64)
+            .def("getParameterBool",   &Region::getParameterBool)
+            .def("getParameterString", &Region::getParameterString)
+            .def("getParameterArray", &Region::getParameterArray);
+
+        py_Region.def("getParameterArrayCount", &Region::getParameterArrayCount);
+
+        py_Region.def("setParameterInt32", &Region::setParameterInt32)
+            .def("setParameterUInt32", &Region::setParameterUInt32)
+            .def("setParameterInt64",  &Region::setParameterInt64)
+            .def("setParameterUInt64", &Region::setParameterUInt64)
+            .def("setParameterReal32", &Region::setParameterReal32)
+            .def("setParameterReal64", &Region::setParameterReal64)
+            .def("setParameterBool",   &Region::setParameterBool)
+            .def("setParameterString", &Region::setParameterString)
+            .def("setParameterArray",  &Region::setParameterArray);
+                
+                
+        py_Region.def("executeCommand", [](Region& r, const std::string& command, py::args args)
```

Comment:
so this is a convenience wrapper for `executeCommand("command" vector<string>)` ?